src/ObjectGraph.es6.js

```diff
@@ -248,12 +276,14 @@ ObjectGraph.prototype.visitObject = function(o) {
 
   var dataMap = this.storeObject(id);
   var metadataMap = this.storeMetadata(id);
+  this.visitPrototype(o, dataMap);
 
-  // Enqueue work: Visit o's prototype and property descriptors.
-  this.q.enqueue(this.visitPrototype.bind(this, o, dataMap));
```

Comment:
Let's keep this and make sure `visitPrototype` uses `maybeSkip` to avoid visiting things multiplet times.